src/System.Runtime/tests/System/String.Split.cs

```diff
@@ -0,0 +1,515 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Xunit;
+
+// TODO: Remove these extension methods when the actual methods are available on String in System.Runtime.dll
+internal static class TemporaryStringSplitExtensions
+{
+    public static string[] Split(this string value, char separator)
+    {
+        return value.Split(new[] { separator });
+    }
+
+    public static string[] Split(this string value, char separator, StringSplitOptions options)
+    {
+        return value.Split(new[] { separator }, options);
+    }
+
+    public static string[] Split(this string value, char separator, int count, StringSplitOptions options)
+    {
+        return value.Split(new[] { separator }, count, options);
+    }
+
+    public static string[] Split(this string value, string separator)
+    {
+        return value.Split(new[] { separator }, StringSplitOptions.None);
+    }
+
+    public static string[] Split(this string value, string separator, StringSplitOptions options)
+    {
+        return value.Split(new[] { separator }, options);
+    }
+
+    public static string[] Split(this string value, string separator, int count, StringSplitOptions options)
+    {
+        return value.Split(new[] { separator }, count, options);
+    }
+}
+
+public static class StringSplitTests
+{
+    [Fact]
+    public static void TestSplitInvalidCount()
+    {
+        const string value = "a,b";
+        const int count = -1;
+        const StringSplitOptions options = StringSplitOptions.None;
+
+        Assert.Throws<ArgumentOutOfRangeException>(() => value.Split(',', count, options));
```

Comment:
Sorry, I edited my above comment for correctness.

You have 6 new overloads in the TemporaryStringSplitExtensions but only 5 assertions in a couple of test cases. I think you're missing a couple of assertions since the number of assertions would be a multiple of 6 per test. In most cases in looks like the tests are missing assertions on one of `Split(separator)` or `Split(separator, options)` or both.